velox/exec/Window.cpp

```diff
@@ -300,8 +305,16 @@ void Window::updateKRowsFrameBounds(
   if (frameArg.index == kConstantChannel) {
     auto constantOffset = frameArg.constant.value();
     auto startValue =
-        startRow + (isKPreceding ? -constantOffset : constantOffset);
-    std::iota(rawFrameBounds, rawFrameBounds + numRows, startValue);
+        (isKPreceding ? -constantOffset : constantOffset) + startRow;
+    for (int i = 0; i < numRows; i++) {
+      // Considers integer overflow case.
+      if (startValue != (int32_t)startValue) {
```

Comment:
Should we do this check for each row ? Since we know 'i' goes from 0 to numRows, startValue is bounded as well. So we know the value of i where the overflow could trigger. We could avoid the if condition in the loop and write 2 for loops instead.